interface/ceed.c

```diff
@@ -919,13 +853,149 @@ int CeedErrorImpl(Ceed ceed, const char *filename, int lineno, const char *func,
   if (ceed) {
     retval = ceed->Error(ceed, filename, lineno, func, ecode, format, args);
   } else {
-    // This function doesn't actually return
     // LCOV_EXCL_START
-    retval = CeedErrorAbort(ceed, filename, lineno, func, ecode, format, args);
+    const char *ceed_error_handler = getenv("CEED_ERROR_HANDLER");
+    if (!strcmp(ceed_error_handler, "return"))
+      retval = CeedErrorReturn(ceed, filename, lineno, func, ecode, format, args);
+    else
+      // This function will not return
+      retval = CeedErrorAbort(ceed, filename, lineno, func, ecode, format, args);
   }
   va_end(args);
   return retval;
   // LCOV_EXCL_STOP
 }
 
+/**
+  @brief Error handler that returns without printing anything.
+
+  Pass this to CeedSetErrorHandler() to obtain this error handling behavior.
+
+  @ref Developer
+**/
+// LCOV_EXCL_START
+int CeedErrorReturn(Ceed ceed, const char *filename, int lineno,
+                    const char *func, int ecode, const char *format,
+                    va_list args) {
+  return ecode;
+}
+// LCOV_EXCL_STOP
+
+/**
+  @brief Error handler that stores the error message for future use and returns
+           the error.
+
+  Pass this to CeedSetErrorHandler() to obtain this error handling behavior.
+
+  @ref Developer
+**/
+// LCOV_EXCL_START
+int CeedErrorStore(Ceed ceed, const char *filename, int lineno,
+                   const char *func, int ecode, const char *format,
+                   va_list args) {
+  if (ceed->parent)
+    CeedErrorStore(ceed->parent, filename, lineno, func, ecode, format, args);
+
+  // Build message
+  char errmsg[CEED_MAX_RESOURCE_LEN], errmsgva[CEED_MAX_RESOURCE_LEN/2];
+  vsprintf(errmsgva, format, args);
+  sprintf(errmsg, "%s:%d in %s(): %s", filename, lineno, func, errmsgva);
+  size_t slen = strlen(errmsg) + 1;
+
+  // Copy message
+  memcpy(ceed->errmsg, errmsg, slen);
```

Comment:
What if `slen` is longer than the buffer? I guess I'd make the buffer have some extra capacity. I think you can do something like this to complexity of buffer management.
```c
  len = snprintf(ceed->errmsg, buffsize, "%s:%d in %s(): ", ...);
  vsnprintf(ceed->errmsg + len, buffsize-len, format, args);
```

Reply:
Agreed. Its less code and safer code. Changed.